### DISCUSSION TITLE
How can I hide my workflow?

### DISCUSSION THREAD

**Discussion by yeyupiaoling** [langgenius/dify#3810] [Suggestion]
I don't want users to see my workflows, I want to hide them. How do I set them?

![20240425103909](https://github.com/langgenius/dify/assets/26297768/7465a86f-e290-4211-a17a-1c73953d789e)

![20240425103944](https://github.com/langgenius/dify/assets/26297768/94fb128f-fed8-4186-b321-d18f01356c37)

**Comment by AndyMik90** [comment]
I suggest we get an option to hide/show this on the front end for the end users. 

There are some apps/assistants that we want to show the workflow process, but in many, we don't.

**Comment by yeyupiaoling** [comment]
Yes! This page is more for users. We developers can view the workflow in studio.

**Comment by BradfordSun** [comment]
I’m currently doing some custom development on it. If you want to hide the workflow, you just need to comment out the following in the file web\app\components\base\chat\chat\answer\index.tsx: 
```
{workflowProcess && (
              <WorkflowProcess data={workflowProcess} hideInfo />
)}
```

**Reply by patryk20120** [reply]
I was thinking about adding setting to show/hide this feature. Will you develop it further?

**Reply by yeyupiaoling** [reply]
@BradfordSun   I see v0.6.8 released. Is there support for setting hidden workflows?

**Reply by toronto19** [reply]
> I’m currently doing some custom development on it. If you want to hide the workflow, you just need to comment out the following in the file web\app\components\base\chat\chat\answer\index.tsx:
> 
> ```
> {workflowProcess && (
>               <WorkflowProcess data={workflowProcess} hideInfo />
> )}
> ```

How can we do it in our local setup, as for running it locally i am creating a virtual env and just doing docker up. can you share the detail step please.

**Comment by crazywoola** [comment]
We will release this feature in next few minor versions.

**Reply by wtslpx** [reply]
Yes,pls add this option.thank you.
but almost one year,nothing happen.
hide workflow node names for end-user is really very important .
thanks for your job.

**Comment by laelhalawani** [comment]
for anyone who thinks the option is not there - it's just real hard to find 😄 
here's how: https://github.com/langgenius/dify/issues/5411#issuecomment-2992539941
